lib/release-tools.nix

```diff
@@ -1,3 +1,23 @@
+{ pkgs ? import <nixpkgs> {} }: 
+
+let
+  nixpkgsPath = pkgs.path;
+
+  # Original `evalConfig`
+  evalConfig' = import "${pkgs.path}/nixos/lib/eval-config.nix";
+
+  # Specialized `evalConfig` which injects the _mobile-nixos special arg.
+  evalConfig = args@{ specialArgs ? {}, ... }: evalConfig' (args // {
```

Comment:
The main difference is how here I have removed the "useless" `eval-config.nix` proxy, and instead made it a function.

This function adds a mostly-internal-use attrset to the arguments, which can be used to

 - refer to the nixpkgs path
 - refer to mobile nixos path
 - re-use this function

Reply:
I'm curious, in most of the cases where you refer to nixpkgs path, it's so that you can reach the modules. Would it be okay to replace those cases with the built-in `modulesPath`?

I've been working on a compatibility shim to get it all working via flakes but I had some issues directly importing `${mobile-nixos}/lib/configuration.nix` since it'll complain infinite recurse from a missing `_mobile-nixos` parameter (I think it's only getting into issues under `_nixos-disintegration/kernel.nix`).

So in that file:

```nix
 toOverride = import "${_mobile-nixos.nixpkgsPath}/nixos/modules/system/boot/kernel.nix" {
```

```nix
 toOverride = import "${modulesPath}/nixos/modules/system/boot/kernel.nix" {
```

Reply:
Indeed, the only leftover place was this location, which I recall needed to use one level of indirection to properly eval (or else infinite recursion).

It does seem to work this way, neat.

Reply:
With that reference gone, you might be able to completely drop `_mobile-nixos` and the custom eval? Nothing seems to reference anything other than `evalConfig` now

Reply:
Something is odd with the specialized `evalConfig`... It is totally required for the `disintegration` parts to happen properly. I wonder if there is something about ordering here going on that we can't trivially control from *inside* the eval.

See the new update, try to make `composeConfig` work without `__mobile-nixos.evalConfig`. I'm a bit at a loss

Testing with:

```
$ nix-instantiate --argstr device razer-cheryl2 -A build.android-recovery
```

When re-importing `evalConfig` from `pkgs.path`, even copying the exact `let ... in` block from `release-tools.nix` I get:

```
error: The option `boot.kernelParams' in `/nix/store/ng1v7167cmf7y2z9kv86dr7dpsx8k7kf-nixpkgs/nixos/modules/system/boot/kernel.nix' is already declared in `/Users/samuel/Projects/mobile-nixos/mobile-nixos/modules/_nixos-disintegration/kernel.nix'.
```